### PR TITLE
Trim down Razor SDK size

### PR DESCRIPTION
**Pull request by captainsafia**
Previous source-generators size: 11.8 MB
New source-generators size: 0.05 MB

### CODE REVIEW

File: src/RazorSdk/SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.csproj [hunk 1]
```diff
@@ -10,9 +10,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.8.0" GeneratePathProperty="true"/>
-    <PackageReference Include="Microsoft.AspNetCore.Razor.Language" Version="$(MicrosoftAspNetCoreRazorLanguageVersion)" GeneratePathProperty="true" />
-    <PackageReference Include="Microsoft.CodeAnalysis.Razor" Version="$(MicrosoftCodeAnalysisRazorVersion)" GeneratePathProperty="true" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.8.0" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.AspNetCore.Razor.Language" Version="$(MicrosoftAspNetCoreRazorLanguageVersion)" PrivateAssets="all"/>
+    <PackageReference Include="Microsoft.CodeAnalysis.Razor" Version="$(MicrosoftCodeAnalysisRazorVersion)" PrivateAssets="all" />
     <PackageReference Include="Microsoft.AspNetCore.Mvc.Razor.Extensions" Version="$(MicrosoftAspNetCoreMvcRazorExtensionsPackageVersion)" GeneratePathProperty="true" />
```
Comment (pranavkm) [inline src/RazorSdk/SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.csproj:16]: This assembly appears under `extensions/mvc-3-0`. We could change that file to be directly under the `tools` directory alongside other rzc assemblies. We don't really have a very good reason (any longer) for it to appear in a different path. It's also only about 70k, so we're not talking about extensive size savings, so I'll let you decide.

File: src/RazorSdk/Targets/Microsoft.NET.Sdk.Razor.SourceGenerators.targets [hunk 2]
```diff
@@ -16,13 +16,14 @@ Copyright (c) .NET Foundation. All rights reserved.
...
       <_RazorSdkSourceGeneratorDirectoryRoot>$(RazorSdkDirectoryRoot)\source-generators\</_RazorSdkSourceGeneratorDirectoryRoot>
+      <_RazorSdkTasksDirectoryRoot>$(RazorSdkDirectoryRoot)\tasks\net6.0\</_RazorSdkTasksDirectoryRoot>
     </PropertyGroup>
 
     <!-- Configure analyzers -->
     <ItemGroup>
-      <_RazorAnalyzer Include="$(_RazorSdkSourceGeneratorDirectoryRoot)Microsoft.AspNetCore.Mvc.Razor.Extensions.dll" />
-      <_RazorAnalyzer Include="$(_RazorSdkSourceGeneratorDirectoryRoot)Microsoft.AspNetCore.Razor.Language.dll" />
-      <_RazorAnalyzer Include="$(_RazorSdkSourceGeneratorDirectoryRoot)Microsoft.CodeAnalysis.Razor.dll" />
+      <_RazorAnalyzer Include="$(_RazorSdkTasksDirectoryRoot)Microsoft.AspNetCore.Mvc.Razor.Extensions.dll" />
```
Comment (pranavkm) [inline src/RazorSdk/Targets/Microsoft.NET.Sdk.Razor.SourceGenerators.targets:24]: I think there's a bunch of extra files in the tasks directory compared to the 5.0 SDK. For e.g, 

```
ls 'C:\Program Files\dotnet\sdk\5.0.103\Sdks\Microsoft.NET.Sdk.Razor\tasks\net5.0'


    Directory: C:\Program Files\dotnet\sdk\5.0.103\Sdks\Microsoft.NET.Sdk.Razor\tasks\net5.0


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----         1/23/2021  12:33 PM          76680 Microsoft.NET.Sdk.Razor.Tasks.dll
```

vs 
```
 ls D:\temp\dotnet6\sdk\6.0.100-preview.3.21124.4\Sdks\Microsoft.NET.Sdk.Razor\tasks\net6.0\


    Directory: D:\temp\dotnet6\sdk\6.0.100-preview.3.21124.4\Sdks\Microsoft.NET.Sdk.Razor\tasks\net6.0

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----         2/24/2021 10:32 AM                cs
d-----         2/24/2021 10:32 AM                de
d-----         2/24/2021 10:32 AM                es
d-----         2/24/2021 10:32 AM                fr
d-----         2/24/2021 10:32 AM                it
d-----         2/24/2021 10:32 AM                ja
d-----         2/24/2021 10:32 AM                ko
d-----         2/24/2021 10:32 AM                pl
d-----         2/24/2021 10:32 AM                pt-BR
d-----         2/24/2021 10:32 AM                ref
d-----         2/24/2021 10:32 AM                ru
d-----         2/24/2021 10:32 AM                tr
d-----         2/24/2021 10:32 AM                zh-Hans
d-----         2/24/2021 10:32 AM                zh-Hant
-a----         2/24/2021  1:15 PM         156672 Microsoft.AspNetCore.Mvc.Razor.Extensions.dll
-a----         2/24/2021  1:15 PM        2572288 Microsoft.AspNetCore.Razor.Language.dll
-a----         2/24/2021  1:15 PM       15308800 Microsoft.CodeAnalysis.CSharp.dll
-a----         2/24/2021  1:15 PM        6086656 Microsoft.CodeAnalysis.dll
-a----         2/24/2021  1:15 PM         174592 Microsoft.CodeAnalysis.Razor.dll
-a----         2/24/2021  1:15 PM         167424 Microsoft.NET.Sdk.Razor.Tasks.dll
```

I would have expected we would to pick up Razor.Language and co from the `tools` directory.

**Comment by captainsafia** [comment]
```.
├── Sdk
│   ├── Sdk.Razor.StaticAssets.ProjectSystem.props
│   ├── Sdk.props
│   └── Sdk.targets
├── build
│   └── netstandard2.0
│       ├── Microsoft.NET.Sdk.Razor.CodeGeneration.targets
│       ├── Microsoft.NET.Sdk.Razor.Compilation.targets
│       ├── Microsoft.NET.Sdk.Razor.Component.targets
│       ├── Microsoft.NET.Sdk.Razor.Configuration.targets
│       ├── Microsoft.NET.Sdk.Razor.DesignTime.targets
│       ├── Microsoft.NET.Sdk.Razor.GenerateAssemblyInfo.targets
│       ├── Microsoft.NET.Sdk.Razor.MvcApplicationPartsDiscovery.targets
│       ├── Microsoft.NET.Sdk.Razor.ScopedCss.targets
│       ├── Microsoft.NET.Sdk.Razor.SourceGenerators.targets
│       ├── Microsoft.NET.Sdk.Razor.StaticWebAssets.targets
│       ├── Microsoft.NET.Sdk.Razor.props
│       ├── Rules
│       │   ├── RazorComponentWithTargetPath.xaml
│       │   ├── RazorConfiguration.xaml
│       │   ├── RazorExtension.xaml
│       │   ├── RazorGeneral.xaml
│       │   └── RazorGenerateWithTargetPath.xaml
│       ├── Sdk.Razor.CurrentVersion.props
│       └── Sdk.Razor.CurrentVersion.targets
├── source-generators
│   ├── Microsoft.NET.Sdk.Razor.SourceGenerators.deps.json
│   └── Microsoft.NET.Sdk.Razor.SourceGenerators.dll
├── targets
│   ├── Microsoft.NET.Sdk.Razor.CodeGeneration.targets
│   ├── Microsoft.NET.Sdk.Razor.Compilation.targets
│   ├── Microsoft.NET.Sdk.Razor.Component.targets
│   ├── Microsoft.NET.Sdk.Razor.Configuration.targets
│   ├── Microsoft.NET.Sdk.Razor.DesignTime.targets
│   ├── Microsoft.NET.Sdk.Razor.GenerateAssemblyInfo.targets
│   ├── Microsoft.NET.Sdk.Razor.MvcApplicationPartsDiscovery.targets
│   ├── Microsoft.NET.Sdk.Razor.ScopedCss.targets
│   ├── Microsoft.NET.Sdk.Razor.SourceGenerators.targets
│   ├── Microsoft.NET.Sdk.Razor.StaticWebAssets.targets
│   ├── Microsoft.NET.Sdk.Razor.props
│   ├── Rules
│   │   ├── RazorComponentWithTargetPath.xaml
│   │   ├── RazorConfiguration.xaml
│   │   ├── RazorExtension.xaml
│   │   ├── RazorGeneral.xaml
│   │   └── RazorGenerateWithTargetPath.xaml
│   ├── Sdk.Razor.CurrentVersion.props
│   └── Sdk.Razor.CurrentVersion.targets
├── tasks
│   ├── net6.0
│   │   ├── Microsoft.NET.Sdk.Razor.Tasks.dll
│   │   └── ref
│   │       └── Microsoft.NET.Sdk.Razor.Tasks.dll
│   └── net472
│       ├── Microsoft.NET.Sdk.Razor.Tasks.dll
│       ├── Microsoft.NET.Sdk.Razor.Tasks.dll.config
│       ├── System.Buffers.dll
│       ├── System.Collections.Immutable.dll
│       ├── System.Memory.dll
│       ├── System.Numerics.Vectors.dll
│       ├── System.Reflection.Metadata.dll
│       └── System.Runtime.CompilerServices.Unsafe.dll
└── tools
    ├── Microsoft.AspNetCore.Mvc.Razor.Extensions.dll
    ├── Microsoft.AspNetCore.Razor.Language.dll
    ├── Microsoft.CodeAnalysis.CSharp.dll
    ├── Microsoft.CodeAnalysis.Razor.dll
    ├── Microsoft.CodeAnalysis.dll
    ├── Microsoft.Css.Parser.dll
    ├── Newtonsoft.Json.dll
    ├── rzc.deps.json
    ├── rzc.dll
    └── rzc.runtimeconfig.json

12 directories, 61 files
```

**Comment by pranavkm** [comment]
Are we able to also eliminate the duplicate files under the build and target directories? Feel free to do this in a follow up if you'd like. Also, if you wanted to squeeze the last bytes out of this, `Microsoft.NET.Sdk.Razor.SourceGenerators.deps.json` isn't required.

**Review by pranavkm — APPROVED** [review]

**Comment by captainsafia** [comment]
> Are we able to also eliminate the duplicate files under the build and target directories?

I think so. I can't recall at the moment why it was duplicated. I seem to recall some test failures but the right fix there would probably be to update things where necessary to resolve from a single directory.

File: src/RazorSdk/SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.csproj [hunk 3]
```diff
@@ -23,9 +23,6 @@
 
   <Target Name="GetDependencyTargetPaths">
     <ItemGroup>
-      <TargetPathWithTargetPlatformMoniker Include="$(PkgMicrosoft_AspNetCore_Razor_Language)\lib\netstandard2.0\Microsoft.AspNetCore.Razor.Language.dll" IncludeRuntimeDependency="false" />
-      <TargetPathWithTargetPlatformMoniker Include="$(PkgMicrosoft_CodeAnalysis_Razor)\lib\netstandard2.0\Microsoft.CodeAnalysis.Razor.dll" IncludeRuntimeDependency="false" />
-      <TargetPathWithTargetPlatformMoniker Include="$(PkgMicrosoft_CodeAnalysis_CSharp)\lib\netstandard2.0\Microsoft.CodeAnalysis.CSharp.dll" IncludeRuntimeDependency="false" />
```
Comment (pranavkm) [inline src/RazorSdk/SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.csproj:28]: I guess everything but this

**Pull request merged**